swarm/src/handler/map_in.rs

```diff
@@ -47,6 +47,7 @@ impl<TConnectionHandler, TMap, TNewIn> MapInEvent<TConnectionHandler, TNewIn, TM
     }
 }
 
+#[allow(deprecated)] // TODO: Remove when {In, Out}boundOpenInfo is fully removed.
```

Comment:
With RFC 2383, `expect` is now stabilized: https://github.com/rust-lang/rust/pull/120924. By using `expect` instead of `allow`, we'll get a warning if the deprecated items have been removed and the `allow`/ `expect` isn't needed anymore.

Question: Is our CI on a recent enough version to support `expect`?

Reply:
(same for all other places where we add `allow(deprecated)` in this PR)

Reply:
> Question: Is our CI on a recent enough version to support `expect`?

It was stabilized in `1.81`, I believe our current workflow is on `1.80`, which may not support this unless they have backported (branched on Jun 7 vs merged on Jun 27) it.

Reply:
Fixed with #5730 :)

Reply:
Well actually, if we do use `expect` we must update our MSRV, see #5650.